test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -803,6 +803,51 @@ TEST_F(HttpFilterTest, AuthWithNonUtf8RequestData) {
   EXPECT_EQ(data_.length(), check_request.attributes().request().http().raw_body().size());
 }
 
+// Checks that the filter buffers the data and initiates the authorization request.
+TEST_F(HttpFilterTest, AuthWithNonUtf8RequestHeaders) {
+  InSequence s;
+
+  // N.B. headers_as_bytes is set to true.
+  initialize(R"EOF(
+  transport_api_version: V3
+  headers_as_bytes: true
+  grpc_service:
+    envoy_grpc:
+      cluster_name: "ext_authz_server"
+  )EOF");
+
+  prepareCheck();
+
+  // Add header with non-UTF-8 value.
+  absl::string_view header_key = "header-with-non-utf-8-value";
+  const uint8_t non_utf_8_bytes[2] = {0xc0, 0xc0};
```

Comment:
Can you add multiple headers with same key but different value to test the above merge duplicate header logic?

Reply:
this is now tested in check_request_utils_test.cc (`HeadersAsBytesNoConcatentation`)